ompi/mca/hook/prot/hook_prot_fns.c

```diff
@@ -0,0 +1,803 @@
+ *  based on whether the previous iteration wrote ".." onto the end.
+ */
+        if (list[i] == hi+1) {
+            hi = list[i];
+        } else if (list[i] > hi) {
+            if (strlen(str)==0 || str[strlen(str)-1] != '.') {
+                if (strlen(str) != 0) {
+                    strcpy(&str[strlen(str)], ", ");
+                }
+                if (lo != hi) {
+                    sprintf(&str[strlen(str)], "%d - %d", lo, hi);
+                } else {
+                    sprintf(&str[strlen(str)], "%d", lo);
+                }
+            }
+/*
+ *  If we've almost written to the end of the string, and we haven't
+ *  already written ".." to indicate we're not writing amy more, then
+ *  add the "..".  Also set hi=lo=i since the data we just wrote is
+ *  for the previous contiguous chunk, and the current i is the start
+ *  of the next chunk.
+ */
+            if (((int)strlen(str)) >= max - 5 - 2*per
+                &&
+                (strlen(str) == 0 || str[strlen(str)-1] != '.'))
+            {
+                strcpy(&str[strlen(str)], ", ..");
+                break;
+            }
+            hi = lo = list[i];
+        }
+    }
+    if (strlen(str)==0 || str[strlen(str)-1] != '.') {
+        if (strlen(str)!=0) {
+            strcpy(&str[strlen(str)], ", ");
+        }
+        if (lo != hi) {
+            sprintf(&str[strlen(str)], "%d - %d", lo, hi);
+        } else {
+            sprintf(&str[strlen(str)], "%d", lo);
+        }
+    }
+}
+
+// Input argument tells where we're being called from:
+// 1 for init, 2 for finalize.
+// The other implicit input is an environment variable we look at.
+// When activated from init: we establish connections before printing.
+// When activated from finalize: we just print whatever info is available.
+static void
+ompi_report_prots(int mode) // 1 = from init, 2 = from finalize
+{
+    int numhosts, i, j, k;
+    char *p;
+    int max2Dprottable = 12;
+    int max2D1Cprottable = 36;
+    int hostidprotbrief = 0;
+    char * max2Dprotptr = NULL;
+    char * hostidprotptr = NULL;
+    int printTable;
+    int hpmp_myrank, hpmp_nprocs;
+    int mylocalrank, nlocalranks, myleaderrank, nleaderranks;
+    int ret;
+    ompi_communicator_t *local_comm, *leader_comm;
+    int *method;
+    char *hoststring;
+    char **allhoststrings;
+
+// early return in the case of spawn
+    // PMPI_Comm_get_parent(&parent);
+    if (ompi_mpi_comm_parent != MPI_COMM_NULL) { return; }
+
+    hpmp_myrank = ompi_comm_rank(MPI_COMM_WORLD);
+    hpmp_nprocs = ompi_comm_size(MPI_COMM_WORLD);
+
+    if (hpmp_myrank == 0) {
+        max2Dprotptr = getenv("MPI_PROT_MAX");
+        if (max2Dprotptr) {
+            max2Dprottable = atoi(max2Dprotptr);
+            max2D1Cprottable = 3 * max2Dprottable;
+        }
+
+        hostidprotptr = getenv("MPI_PROT_BRIEF");
+        if (hostidprotptr) { hostidprotbrief = atoi(hostidprotptr); }
```

Comment:
@markalle It looks like `MPI_PROT_BRIEF` value is not used. Was the intention to force the "brief" output for all -prot output?